tests/ArgTest.php

```diff
@@ -805,7 +805,7 @@ function caller($callback) {}
                     function foo($a) {}
 
                     caller("foo");',
-                'error_message' => 'InvalidScalarArgument',
+                'error_message' => 'PossiblyInvalidArgument',
```

Comment:
Not sure why this could be considered possibly invalid

Reply:
This seems to be caused by a call to `UnionTypeComparator::canBeContainedBy`, which doesn't have the same callable parameter comparison logic that `UnionTypeComparator::isContainedBy` has.

Actually, that logic should probably be moved to the `CallableTypeComparator`...

Reply:
Ah I see now, it's to allow things like https://psalm.dev/r/3cc38f5a0c